packages/cursorless-vscode/src/ide/vscode/hats/VscodeHatRenderer.ts

```diff
@@ -99,12 +104,52 @@ export default class VscodeHatRenderer {
     return this.decorationMap[hatStyle];
   }
 
+  private async updateHatsDirWatcher() {
+    this.destroyHatsDirWatcher();
+
+    const hatsDir = vscode.workspace
+      .getConfiguration("cursorless.experimental")
+      .get<string>("hatShapesDir")!;
+
+    if (hatsDir && fs.existsSync(hatsDir)) {
+      await this.updateShapeOverrides(hatsDir);
+      let timeout: NodeJS.Timeout;
+
+      this.hatsDirWatcher = fs.watch(hatsDir, () => {
+        clearTimeout(timeout);
+        timeout = setTimeout(() => this.updateShapeOverrides(hatsDir), 50);
```

Comment:
I wonder if we should bake debounce into our watcher api. In my other PR, I also modify our debouncer to accept a timeout, so you could use that if I merge that piece in. I don't love reinventing the wheel in our codebase

Reply:
Sounds reasonable

Reply:
Should we use ide() for things that are in vscode outside of the Cursorless engine?

Reply:
After switching to vscode file watcher I had no need of debouncing any more.